File: lab5/python/greedy_kp.py
import sys

from knapsack import knapsack

class greedy(knapsack):
    def __init__(self, filename):
        knapsack.__init__(self, filename)
        
    def greedy(self):
        self.total_weight=0 # current total weight of the items in the knapsack
        self.total_value=0 # current total profit of the items in the knapsack
        i = 1
        solution = [None]*(self.Nitems + 1)
        
        self.sort_by_ratio() # sort by profit-to-weight ratio
        
        # ADD CODE HERE TO COMPUTE THE GREEDY SOLUTION
        
        # THE CODE SHOULD: take the items in descending
        # profit-to-weight ratio order (by using temp_indexes) and,
        # if an item fits, add it to the knapscak, and
        # do not stop at the first item that doesn't fit
        # - but keep going until all items have been tried.
        
        print("The greedy solution - not necessarily optimal - is:")
        self.check_evaluate_and_print_sol(solution)
        
        # NOTE: If the result to you get when you use the check_ ...() function
        # is not what you expected, if could be because you mapped to the sorted
        # order TWICE
        # Use
        #   solution[i] = Trye
        # in order to "pack" the iths most value-dense item,
        # not solution[temp_indexes[i]]
        
knapsk = greedy(sys.argv[1])
knapsk.print_instance()
knapsk.greedy()

File: lab5/java/comp26120/greedy_kp.java
package comp26120;

import java.util.ArrayList;

public class greedy_kp extends KnapSack {

    public greedy_kp(String filename) {
	super(filename);
    }

    public static void main(String[] args) {
	greedy_kp knapsack = new greedy_kp(args[0]);
	knapsack.print_instance();
	knapsack.greedy();
    }

    public void greedy() {
	int total_weight=0; // current total weight of the items in the knapsack
	int total_value=0; // current total profit of the items in the knapsack
	int i;
	ArrayList<Boolean> solution = new ArrayList<Boolean>(Nitems+1);
	// initialise the knapsack initially empty
        for (i = 0; i <= Nitems; i++) {
            solution.add(null);
        }
	i = 1;

	sort_by_ratio(); //sort items in descending profit-to-weight ratio order

	/* ADD CODE HERE TO COMPUTE THE GREEDY SOLUTION */

	/* THE CODE SHOULD: take the items in descending 
	   profit-to-weight ratio order (by using temp_indexes) and, 
	   if an item fits, add it to the knapsack, and 
	   do not stop at the first item that doesn't fit 
	   - but keep going until all items have been tried */

	System.out.println("The greedy solution - not necessarily optimal - is:");
	check_evaluate_and_print_sol(solution);

	 /* NOTE: If the result you get when you use the check_ ...() function
           is not what you expected, it could be because you mapped
           to the sorted order TWICE.
           Use 
              solution[i]=1; 
           in order to "pack" the ith most value-dense item, 
           not solution[temp_index[i]];
	 */


    }
}
